catalog/ui/catalog-ui-filter/src/main/java/org/codice/ddf/catalog/ui/filter/impl/json/JsonFunctionRegistry.java

```diff
@@ -0,0 +1,119 @@
+
+  @Deprecated private static final String TEMPLATE_VALUE_V1 = "template.value.v1";
+
+  public List<JsonFunctionTransform> getTransforms() {
+    return TRANSFORMS;
+  }
+
+  private static class TemplatePropertiesTransform implements JsonFunctionTransform {
+
+    @Override
+    public boolean canApplyToFunction(Map<String, Object> customJson) {
+      notNull(customJson);
+      return customJson.containsKey(FilterJson.Keys.TEMPLATE_PROPS);
+    }
+
+    @Override
+    public boolean canApplyFromFunction(Map<String, Object> filterJson) {
+      notNull(filterJson);
+      Object val = filterJson.get(FilterJson.Keys.VALUE);
+      return val instanceof Map
+          && isFunction((Map) val)
+          && TEMPLATE_VALUE_V1.equals(getFuncName((Map) val));
+    }
+
+    @Override
+    public Map<String, Object> toFunction(Map<String, Object> customJson) {
+      notNull(customJson);
+      Map<String, Object> templateProperties = getTemplateProperties(customJson);
+      return ImmutableMap.<String, Object>builder()
+          .put(FilterJson.Keys.TYPE, customJson.get(FilterJson.Keys.TYPE))
+          .put(FilterJson.Keys.PROPERTY, customJson.get(FilterJson.Keys.PROPERTY))
+          .put(
+              FilterJson.Keys.VALUE,
+              ImmutableMap.<String, Object>builder()
+                  .put(FilterJson.Keys.TYPE, FilterJson.Ops.FUNC)
+                  .put(FilterJson.Keys.NAME, TEMPLATE_VALUE_V1)
+                  .put(
+                      "args",
+                      keyValuesToOrderedList(
+                          templateProperties, "defaultValue", "nodeId", "isVisible", "isReadOnly")))
+          .build();
+    }
+
+    @Override
+    public Map<String, Object> fromFunction(Map<String, Object> filterJson) {
+      notNull(filterJson);
+      List<Object> params = getParams((Map) filterJson.get(FilterJson.Keys.VALUE));
+      return ImmutableMap.<String, Object>builder()
+          .put(FilterJson.Keys.TYPE, filterJson.get(FilterJson.Keys.TYPE))
+          .put(FilterJson.Keys.PROPERTY, filterJson.get(FilterJson.Keys.PROPERTY))
+          .put(
+              FilterJson.Keys.TEMPLATE_PROPS,
+              orderedListToMap(params, "defaultValue", "nodeId", "isVisible", "isReadOnly"))
+          .build();
+    }
+
+    private List<Object> keyValuesToOrderedList(Map<String, Object> src, String... keys) {
+      notNull(src);
+      notEmpty(keys);
+      return Arrays.stream(keys).map(src::get).collect(Collectors.toList());
+    }
+
+    private Map<String, Object> orderedListToMap(List<Object> src, String... keys) {
+      notNull(src);
+      notEmpty(keys);
+      if (src.size() != keys.length) {
+        throw new IllegalArgumentException(
+            String.format(
+                "List of objects (%s) should map 1-to-1 with keys (%s)",
+                src.toString(), Arrays.toString(keys)));
+      }
+      Map<String, Object> result = new HashMap<>();
```

Comment:
❓ Don't you want to retain order here? If so, should you use a `LinkedHashMap` instead?
✏️ This is quickly hacked together (using Guava, and even there it's difficult because both the Java and Guava teams refuse to provide basic collections functionality such as `zip` and tuples), but is close to what you have. All it's missing is the `notNull` check on the keys, but that could easily be accomplished with an in-process or pre-process check.
```java
return Streams.zip(Arrays.stream(keys), list.stream(), Maps::immutableEntry)
        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```
Were you to go with this solution and wanted to retain insertion order, you'd need to provide a map supplier to the collect method.